### PR TITLE
[RPS-262] Introduce Gatling for load tests

### PR DESCRIPTION
**Pull request by wojtek-oledzki**
Also added travis - slack integration and updated github repository
names for CodePipeline.

### CODE REVIEW

File: ansible/roles/foundations/aws_cd/files/code_pipeline_ps_hippo.yml [hunk 1]
```diff
@@ -101,14 +101,14 @@ Resources:
...
         NamespaceType: NONE
-        Path: ps-hippo
+        Path: hippo
         Packaging: ZIP
-      Description: Package ps-hippo repo for other builds
+      Description: Package hippo repo for other builds
       Environment:
         ComputeType: BUILD_GENERAL1_SMALL
-        Image: woledzki/alpine-virtualenv:latest
+        Image: selenium/standalone-chrome:latest
```
Comment (ivorc) [inline ansible/roles/foundations/aws_cd/files/code_pipeline_ps_hippo.yml]: What's the implication of this? This seems unrelated to the other changes. Should it be in this PR?

File: gatling/Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		$(MAKEFILE_LIST)
+
+## Starts Gatling Recorder
+# Usage: make run.recorder
+run.recorder: vendor/recorder
+	bash vendor/recorder -of $(PWD)/simulations/
+
+## Starts Gatling
+# Usage: make run.gatling
+run.gatling: vendor/gatling
```
Comment (ivorc) [inline gatling/Makefile]: What about renaming this to just `test`?

File: gatling/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	@awk -v skip=1 \
+		'/^##/ { sub(/^[#[:blank:]]*/, "", $$0); doc_h=$$0; doc=""; skip=0; next } \
+		 skip  { next } \
+		 /^#/  { doc=doc "\n" substr($$0, 2); next } \
+		 /:/   { sub(/:.*/, "", $$0); printf "\033[34m%-30s\033[0m\033[1m%s\033[0m %s\n\n", $$0, doc_h, doc; skip=1 }' \
+		$(MAKEFILE_LIST)
+
+## Starts Gatling Recorder
+# Usage: make run.recorder
+run.recorder: vendor/recorder
```
Comment (ivorc) [inline gatling/Makefile]: What about renaming this to just `record`?

File: gatling/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Gatling
+
+More information about Gatling tool can be found here: https://gatling.io/
+
+* `make run.gatling` runs Gatling
```
Comment (ivorc) [inline gatling/README.md]: Could maybe just expand a tiny bit on what "runs Gatling" means. Something like "ONLY RUN IF YOU WANT TO PERFORM A LOAD TEST AGAINST THE CONFIGURED ENVIRONMENT. Runs Gatling load injector for performance and capacity testing of the configured environment."

File: gatling/simulations/LoadTest.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.concurrent.TimeUnit
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import io.gatling.jdbc.Predef._
+
+class LoadTest extends Simulation {
+
+	val httpProtocol = http
+		.baseURL("http://hippo-delivery.web.dev.ps.nhsd.io")
```
Comment (ivorc) [inline gatling/simulations/LoadTest.scala]: Is it worth pulling this out as a configuration param, even if just as a local const to highlight that it may want to be changed?

File: gatling/simulations/LoadTest.scala [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+		publicationSystem.scn.inject(
+			rampUsers(150) over(60 seconds),
+			rampUsers(300) over(60 seconds),
+			rampUsers(500) over(60 seconds),
+			rampUsers(800) over(60 seconds),
+			rampUsers(1200) over(60 seconds),
+			rampUsers(1600) over(60 seconds),
+			rampUsers(2000) over(60 seconds)
+		),
+		downloads.scn.inject(
```
Comment (ivorc) [inline gatling/simulations/LoadTest.scala:32]: Does this do the downloads after the rest of the user journey? Don't we want to include the downloads as part of the user journey?

**Review by ivorc — COMMENTED** [review]
I like it, but have a few comments and questions.

File: gatling/simulations/LoadTest.scala (hunk 6 above)
Comment (wojtek-oledzki) [reply gatling/simulations/LoadTest.scala:32]: all scenarios runs at the same time.
